sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClient.java

```diff
@@ -592,6 +597,47 @@ public GlobalThroughputControlConfigBuilder createGlobalThroughputControlConfigB
         return new GlobalThroughputControlConfigBuilder(this, databaseId, containerId);
     }
 
+    void openConnectionsAndInitCaches() {
+        blockListVoidResponse(openConnectionsAndInitCachesInternal());
+    }
+
+    private Mono<List<Void>> openConnectionsAndInitCachesInternal() {
+        int concurrency = 1;
+        int prefetch = 10;
+        if (this.proactiveContainerInitConfig != null) {
+            return Flux.fromIterable(this.proactiveContainerInitConfig.getCosmosContainerIdentities())
+                    .flatMap(
+                        cosmosContainerIdentity -> Mono.just(this
+                            .getDatabase(containerIdentityAccessor.getDatabaseName(cosmosContainerIdentity))
+                            .getContainer(containerIdentityAccessor.getContainerName(cosmosContainerIdentity))),
+                        concurrency,
+                        prefetch
+                    )
+                    .flatMap(
+                        cosmosAsyncContainer -> cosmosAsyncContainer
+                            .openConnectionsAndInitCaches(
+                                this.proactiveContainerInitConfig.getNumProactiveConnectionRegions()),
+                        concurrency,
+                        prefetch)
+                    .collectList();
+        }
+        return Mono.empty();
+    }
+
+    private void blockListVoidResponse(Mono<List<Void>> voidListMono) {
+        try {
+            voidListMono.block();
+        } catch (Exception ex) {
+            final Throwable throwable = Exceptions.unwrap(ex);
+
+            if (throwable instanceof CosmosException) {
```

Comment:
why the different handling of exceptions here

Reply:
Throwable might not be a RuntimeException - so, we need to map it into a Runitme exception - otherwise we would need to call out the exact exceptions the method can throw.

Reply:
Won't fix/change